src/ontology/uberon.Makefile

```diff
@@ -90,7 +90,7 @@ checks: $(REPORTDIR)/uberon-edit-xp-check $(REPORTDIR)/uberon-edit-obscheck.txt
 # TODO: Huge number of printouts that pollute the general logs
 $(OWLSRC): $(SRC) $(COMPONENTSDIR)/disjoint_union_over.ofn $(REPORTDIR)/$(SRC)-gocheck $(REPORTDIR)/$(SRC)-iconv $(SCRIPTSDIR)/expand-dbxref-literals.pl
 	echo "STRONG WARNING: issues/contributor.owl needs to be manually updated."
-	owltools --no-logging $(UCAT) $< $(COMPONENTSDIR)/disjoint_union_over.ofn issues/contributor.owl --merge-support-ontologies --expand-macros -o  $@.tmp &&  $(SCRIPTSDIR)/expand-dbxref-literals.pl $@.tmp
+	owltools --no-logging $(UCAT) $< $(COMPONENTSDIR)/disjoint_union_over.ofn issues/contributor.owl --merge-support-ontologies --expand-macros -o  $@ &&  $(SCRIPTSDIR)/expand-dbxref-literals.pl $@ > $@.tmp
 	$(ROBOT) query -i $@.tmp --update $(SPARQLDIR)/taxon_constraint_never_in_taxon.ru --update $(SPARQLDIR)/remove_axioms.ru -o $@
```

Comment:
Seems the update query (row 94, using $@.tmp) may not be correctly applied if row 93 results in $@?

Reply:
Before the patch, the `owltools` command in line 93 would send its result to `$@.tmp`. The `expand-dbxref-literals.pl` script (on the same line, after the `&&`) would then read that `$@.tmp` file, and send its own result to the standard output (i.e. on the terminal), where it is not used at all (there’s no redirection to capture the output into a file and use it later). Then on line 94, the update query would read again the output of the previous `owltools` command (still in `$@.tmp`). In other words, `expand-dbxref-literals.pl` is invoked for nothing at all, its output is lost.

With the patch applied, the `owltools` command in line 93 will now send its result to `$@`. Those results will then be read by the `expand-dbxref-literals.pl` script (on the same line), which will send its own result to `$@.tmp`. Then in line 94 the update query will work from the result on the `expand-dbxrefs-literals.pl` taken from `$@.tmp`.

I would have preferred to pipe the output of the `owltools` command directly to the `expand-dbref-literals.pl` script (_i.e._, `owltools ... | $(SCRIPTSDIR)/expand-dbxref-literals.pl > $@.tmp`), but unfortunately it seems that `owltools` does not allow to send its output to stdout, only to a real file (the classical Unix-y convention of using `-` to represent stdout does not work).

Reply:
Thanks @gouttegd all great.